src/core/geometry_func_unittest.cpp

```diff
@@ -0,0 +1,32 @@
+/* $Id$ */
+
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file geometry_func_unittest.cpp Geometry functions unit tests. */
+
+#include "../stdafx.h"
+#include "geometry_func.hpp"
+#include "math_func.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(MaxdimTest, Zero) {
+	Dimension d1 = {0, 0};
+	Dimension d2 = {120, 100};
+
+	EXPECT_EQ(120, maxdim(d1, d2).width);
```

Comment:
What actually does "EXPECT_EQ" do, especially if the two expressions don't equal each other?

Reply:
`EXPECT_*` macros are similar to `ASSERT_*` in the sense that they both test some condition and generate a failure if this condition fails. The difference between `ASSERT_*` and `EXPECT_*` is that when using `ASSERT_*` the function will be aborted on failure, while `EXPECT_*` will not abort the function. The `_EQ` part means that it will test if both arguments are equal.

In this specific case, if those expressions do not equal google test will report the test as failed, but the test is not aborted on the failure.